### ISSUE TITLE
Tx Pre-Sequencing

### ISSUE THREAD

**Issue by jaekwon** [tendermint/tendermint#1168]
A Sequencer is a signer who orders byteslices by signing it with some meta data:

`signBytes := []byte{<chainID/height/txIndex/txBytes>}`

And communicated in a message...

```golang
type TxSequence struct {
  Sequencer crypto.Addr
  ChainID string
  Height int64
  TxIndex int32
  Txs [][]byte
  Gas int64
}
type TxSequenceSigned struct {
  TxSequence
  crypto.Signature
}
```

And so a TxSequenceSigned is self-certified, assuming you had the pubkey.

So the proposer can go ahead and broadcast it, and if peers allowed for this early broadcast perhaps by up to 1 height (for round 0) or 1 round (for round 1+)... and so as to not disrupt the voting that must happen, this early broadcasting of TxSequences would happen with lower priority than the rest of consensus.

Kinda like BitcoinNG, but even before the work.

**Comment by jaekwon** [comment]
For anyone who wants to dive into Tendermint, implementing this feature would be absolutely the wrong way to do it.

**Comment by creachadair** [comment]
@cmwaters, @marbar3778 Do you know if this issue is still relevant? I don't entirely understand the proposal, but it sounds similar in intent to what will happen with PrepareProposal in ABCI++ (though perhaps more specific).

**Comment by cmwaters** [comment]
To me it sounds exactly what a proposer would do to propose a block (sign over a set of transactions, a chainID and a height). I'm not sure if this was intended to operate in parallel with the block proposal process nor what use case is was meant to serve. I think it's fairly safe to close this. If anyone disagrees, then feel free to reopen it and we can discuss it further.
